core/chains/evm/client/pool.go

```diff
@@ -195,36 +196,41 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
-	nodes := p.liveNodes()
-	nNodes := len(nodes)
-	if nNodes == 0 {
-		p.logger.Critical("No live RPC nodes available")
-		return &erroringNode{errMsg: fmt.Sprintf("no live nodes available for chain %s", p.chainID.String())}
+func (p *Pool) getBestNode() Node {
+	highestHeadNumber := int64(-1)
+	var node Node
+	for _, n := range p.nodes {
+		latestReceivedBlockNumber := n.LatestReceivedBlockNumber()
+		if n.State() == NodeStateAlive && n.LatestReceivedBlockNumber() > highestHeadNumber {
+			node = n
+			highestHeadNumber = latestReceivedBlockNumber
+		}
 	}
 
-	// NOTE: Inc returns the number after addition, so we must -1 to get the "current" counter
-	count := p.roundRobinCount.Inc() - 1
-	idx := int(count % uint32(nNodes))
-
-	return nodes[idx]
-}
-
-func (p *Pool) liveNodes() (liveNodes []Node) {
-	for _, n := range p.nodes {
-		if n.State() == NodeStateAlive {
-			liveNodes = append(liveNodes, n)
+	p.lastBestNodeMu.Lock()
+	if p.lastBestNode != nil {
+		if p.lastBestNode.LatestReceivedBlockNumber() >= highestHeadNumber {
```

Comment:
We must be sure that this mutex doesn't block for a long time.

Reply:
IIUC the logic here finds one of the best nodes and then does an additional check to see if we need to actually change or be sticky to our last best, correct?
Can you just initialize highestHeadNumber := p.lastBestNode.LatestReceivedBlockNumber() and then it will be overwritten only by a strictly better node?